src/Http/Http/src/Internal/DefaultHttpRequest.cs

```diff
@@ -147,10 +147,8 @@ public override IRequestCookieCollection Cookies
         set { Headers.ContentType = value; }
     }
 
-    public override bool HasFormContentType
-    {
-        get { return FormFeature.HasFormContentType; }
-    }
+    public override bool HasFormContentType => HttpExtensions.HasAnyFormContentType(ContentType)
+        || _features.Fetch(ref _features.Cache.Form, _features.Collection, collection => collection.Get<IFormFeature>()) != null;
```

Comment:
Isn't this just checking for the presence of the form feature? If present, it should check the form feature's HasFormContentType.

The current form feature should also take precedence if it exists.